### PR TITLE
Use PHP constants for doctrine config

### PR DESCRIPTION
**Pull request by matks**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.2.x
| Description?      | Use PHP constants for doctrine config as informations collected in https://github.com/PrestaShop/PrestaShop/issues/33524 indicate the value of `PDO::MYSQL_ATTR_MULTI_STATEMENTS` change on different environments . This PR is targeting `8.2.x` instead of `develop` and replaces https://github.com/PrestaShop/PrestaShop/pull/36732
| Type?             | bug fix
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Please see below
| UI Tests          | I will add the UI tests when I have fixed the CI
| Fixed issue or discussion?     |  Should fix https://github.com/PrestaShop/PrestaShop/issues/33524
| Related PRs       | 
| Sponsor company   | 

### How to test

As this is a config specific item, the behavior of the shop should be unchanged. To verify the item, a developer could do:

1. Take `develop` branch and insert the below 2 lines in a Symfony controller to be able to verify the value of the configuration options passed to Doctrine
2. Take this Pull Request branch and do the same, verify that despite the code changes the configuration options passed to Doctrine remain unchanged

2 lines of code
```
$b = $this->get('doctrine.dbal.default_connection');
dump($b->getParams());
```

The behavior is "nothing has changed" because I believe the fix will only fix the situation for a subset of users, victims of https://github.com/PrestaShop/PrestaShop/issues/33524

### CODE REVIEW

**Review by kpodemski — APPROVED** [review]

**Review by marsaldev — APPROVED** [review]
LGTM 👍🏼

**Comment by kpodemski** [comment]
@PrestaShop/qa-functional no functional tests are needed in my opinion, installation process is tested multiple times in automated tests

**Review by boherm — APPROVED** [review]

**Review by nicosomb — APPROVED** [review]

**Comment by florine2623** [comment]
Ok @kpodemski , I agree :)

**Comment by ps-jarvis** [comment]
PR merged, well done!

Message to @PrestaShop/committers: do not forget to milestone it before the merge.

**Comment by kpodemski** [comment]
thank you @matks !

**Comment by jf-viguier** [comment]
@matks @kpodemski is this PR has to be ported in 9.x ?

**Comment by kpodemski** [comment]
@jf-viguier I think it should, indeed, be also part of v9, it seems `8.2.x` branch wasn't correctly upmerged

**Comment by kpodemski** [comment]
@jf-viguier could you send a PR?

**Comment by jf-viguier** [comment]
@kpodemski I don't know, this change is require for php 8.5 so It's not so easy.

**Comment by kpodemski** [comment]
@jf-viguier what do you mean? It works in 8.2.x, why wouldn't this work in v9? what would require PHP 8.5?

**Comment by jf-viguier** [comment]
@kpodemski I mean @Progi1984 reverted this change for php 8.5 support here : https://github.com/PrestaShop/PrestaShop/commit/47b386c60c3d5acf3f670db319f6ca282d03189a#diff-c7555cd93894315af6318a1ff369014db27c3db344f1ee5c514c696c6d5eba18

**Pull request merged**
